File: Data Structures & Algorithms/buy-and-sell-crypto/submission-6.java
class Solution {
    public int maxProfit(int[] prices) {
        int max = 0;
        int l = 0;
        int r = 1;
        while(r < prices.length)
        {
            max = Math.max(max, prices[r] - prices[l]);
            if(prices[l] > prices[r])
            {
                l = r;
            }
            r += 1;
        }
        return max;
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-6.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l = 1
        r = max(piles)
        curr = r
        while l <= r:
            mid = l + (r - l) // 2
            ct = 0
            for x in piles:
                ct += math.ceil(x / mid)
            if ct > h:
                l = mid + 1
            else:
                r = mid - 1
                curr = min(curr, mid)
        return curr

       
       
       
       
       
       
       
       
       
       
       
       
       
       
       
       
       
       
       
        # l = 1
        # r = max(piles)
        # curr = r
        # while l <= r:
        #     mid = l + (r - l) // 2
        #     tot = 0
        #     for x in piles:
        #         tot = tot + math.ceil(x / mid)
        #     print(mid)
        #     print(tot)
        #     print(l)
        #     print(r)
        #     if tot > h :
        #         l = mid + 1
        #     else:
        #         curr = min(curr, mid)
        #         r = mid - 1
        # return curr